### ISSUE TITLE
Add halfChecked controlled state in rowSelection for Table with tree structured dataSource (indeterminate checkbox) 

### ISSUE THREAD

**Issue by pepe95270** [ant-design/ant-design#39778]
### What problem does this feature solve?

For now, if we have a Table with tree structured dataSource, there is no way to manage the `indeterminate` state of the checkboxes.
In `rowSelection` of the Table component, the `selectedRowKeys` prop doesn't allow to manage halfChecked keys like we can in the Tree component with `checkedKeys` prop.
Moreover, if we try to manually change the checkbox prop to `indeterminate` with `rowSelection.getCheckboxProps`, it doesn't work and a warning is shown : `Warning: [antd: Table] set 'indeterminate' using 'rowSelection.getCheckboxProps' is not allowed with tree structured dataSource.`

This can be an absolute necessity when using tree structured dataSource with `checkStrictly = true`.

### What does the proposed API look like?

Like for the Tree component, when `checkStrictly = true` , `selectedRowKeys` should look like `{checked: string[], halfChecked: string[]}`

**Comment by FanFanFantastic** [comment]
> ### What problem does this feature solve?
> For now, if we have a Table with tree structured dataSource, there is no way to manage the `indeterminate` state of the checkboxes. In `rowSelection` of the Table component, the `selectedRowKeys` prop doesn't allow to manage halfChecked keys like we can in the Tree component with `checkedKeys` prop. Moreover, if we try to manually change the checkbox prop to `indeterminate` with `rowSelection.getCheckboxProps`, it doesn't work and a warning is shown : `Warning: [antd: Table] set 'indeterminate' using 'rowSelection.getCheckboxProps' is not allowed with tree structured dataSource.`
> 
> This can be an absolute necessity when using tree structured dataSource with `checkStrictly = true`.
> 
> ### What does the proposed API look like?
> Like for the Tree component, when `checkStrictly = true` , `selectedRowKeys` should look like `{checked: string[], halfChecked: string[]}`

I run into the same issue recently, upgrading from antd3 to antd4 . is there any update on this issue?

**Comment by FanFanFantastic** [comment]
> 

I set checkStrictly: false on rowSelection property in the table, this seems to solve the problem? it does show half checked when clicking on the child

**Comment by pepe95270** [comment]
> > 
> 
> I set checkStrictly: false on rowSelection property in the table, this seems to solve the problem? it does show half checked when clicking on the child
Yes but `checkStrictly = false` implies that you can't control manually the state of the parent checkbox, so unfortunately this is not a good solution for me.

@07akioni Is there a particular reason why we can't use `indeterminate` with `rowSelection.getCheckboxProps` for tree structured dataSource ?

**Comment by andriiDatsiuk** [comment]
I've just noticed that because it seems like half checked state is not displayed for checkboxes with children when they are disabled using `getCheckboxProps`

@pepe95270 I was close. Great, thank you!
